client/src/components/Cart/Cart.tsx

```diff
@@ -23,20 +23,20 @@ function Cart({
   return (
     <div className="Cart">
       {contents.length > 0 &&
-        contents.map((cartItem) =>
-          collapse ? (
-            <ItemCardCompact cartItem={cartItem} />
+        contents.map((cartItem) => {
+          const cardProps = {
+            key: cartItem.item.barcode,
+            cartItem: cartItem,
+            updateItemQuantity: updateItemQuantity
+              ? updateItemQuantity
+              : (barcode: string, quantity: number) => {},
```

Comment:
Does this throw error? If its just an anonymous function, it can be shortened. Also aren't you already setting a default value in the function paramter for this? Checking it again is not required.
```suggestion
              : ({}) => {},
```

Reply:
unfortunately, no default value is defined for undefined props (updateItemQuantity is an optional prop). Rationale: we may be reusing `<Cart />` as our receipt, in which case, a collapsed-only Cart will not require this updateItemQuantity callback.
However, anonymous function can be declared and passed as `() => {}`, I though typescript would complain but looks like that is not the case :)